app/controllers/hyrax/handles_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Hyrax
+  class HandlesController < ApplicationController
+    def handle_show
+      if params[:handle_id] != "1957"
+        puts "Handle ID does not match with Scholars Archive Handle ID."
+      elsif params[:id].length != 5
+        puts "ID is not the right length for Scholars Archive Handle ID."
+      end
+
+      solr_doc = query_solr_for_work(construct_handle_url(params[:id]))
+      work = query_fedora_for_work(solr_doc.first[:id], solr_doc.first[:human_readable_type_tesim].first.constantize)
+      redirect_to "/concern/#{work.class.to_s.downcase}s/#{work.id}"
```

Comment:
I feel like you should use `pluralize` or look at the Rails doc to see if there is a "class to plural string" type of a method. Either way, adding `s` to the end of the class name isn't ever safe for trying to pluralize word.